### PR TITLE
Fix/blocklist

### PR DESCRIPTION
**Pull request by BLuEScioN**
closes [#1603](https://github.com/hirosystems/explorer/issues/1603)

### CODE REVIEW

File: src/app/_components/BlockList/Grouped/BlockListGrouped.tsx [hunk 1]
```diff
@@ -260,29 +257,30 @@ export function Footer({ btcBlock, txSum }: { btcBlock: BlockListBtcBlock; txSum
...
 }) {
-  const numStxBlocks = btcBlock.txsCount ?? stxBlocks.length;
-  const numStxBlocksNotDisplayed = numStxBlocks - (stxBlocksLimit || 0);
-  const txSum = stxBlocks.reduce((txSum, stxBlock) => {
-    const txsCount = stxBlock?.txsCount ?? 0;
-    return txSum + txsCount;
-  }, 0);
-  const stxBlocksShortList = stxBlocksLimit ? stxBlocks.slice(0, stxBlocksLimit) : stxBlocks;
+  const unaccountedStxBlocks = stxBlocks.length - btcBlock.blockCount;
+  const unaccountedTxs = stxBlocks
```
Comment (He1DAr) [inline src/app/_components/BlockList/Grouped/BlockListGrouped.tsx:271]: could benefit from useMemo

File: src/app/_components/BlockList/Grouped/BlockListGrouped.tsx [hunk 2]
```diff
@@ -260,29 +257,30 @@ export function Footer({ btcBlock, txSum }: { btcBlock: BlockListBtcBlock; txSum
...
+    .slice(0, unaccountedStxBlocks)
+    .reduce((acc, block) => acc + (block.txsCount || 0), 0);
+  const txsCount = isFirst ? btcBlock.txsCount + unaccountedTxs : btcBlock.txsCount;
+  const blocksCount = isFirst ? btcBlock.blockCount + unaccountedStxBlocks : btcBlock.blockCount;
+  const numStxBlocksNotDisplayed = blocksCount - (stxBlocksLimit || 0);
+  const displayedStxBlocks = stxBlocksLimit ? stxBlocks.slice(0, stxBlocksLimit) : stxBlocks;
```
Comment (He1DAr) [inline src/app/_components/BlockList/Grouped/BlockListGrouped.tsx:277]: same for this one (useMemo)

File: src/common/queries/useBurnBlock.ts [hunk 3]
```diff
@@ -8,9 +8,20 @@ import {
...
+import { useGlobalContext } from '../context/useAppContext';
 
 export const BURN_BLOCKS_QUERY_KEY = 'burnBlocks';
 
+export function useGetBurnBlockQuery() {
+  const { url: activeNetworkUrl } = useGlobalContext().activeNetwork;
+  return (heightOrHash: string | number) => ({
+    queryKey: [BURN_BLOCKS_QUERY_KEY, heightOrHash],
+    queryFn: () =>
+      fetch(`${activeNetworkUrl}/extended/v2/burn-blocks/${heightOrHash}`).then(res => res.json()),
```
Comment (He1DAr) [inline src/common/queries/useBurnBlock.ts:20]: Here also we can use BurnBlocksApi.getBurnBlock

File: src/app/_components/BlockList/utils.ts [hunk 4]
```diff
@@ -57,32 +54,33 @@ export function waitForFadeAnimation(callback: () => void) {
...
+        : createBlockListBtcBlock(firstStxBlock, firstBtcBlock),
     },
   ];
+
   if (stxBlocks.length === 1) return blockList;
+
   for (let i = 1; i < stxBlocks.length; i++) {
     const stxBlock = stxBlocks[i];
     const latestBtcBlock = blockList[blockList.length - 1].btcBlock;
     const latesStxBlocks = blockList[blockList.length - 1].stxBlocks;
```
Comment (He1DAr) [inline src/app/_components/BlockList/utils.ts:76]: typo

File: src/app/_components/BlockList/utils.ts [hunk 5]
```diff
@@ -57,32 +54,33 @@ export function waitForFadeAnimation(callback: () => void) {
   setTimeout(callback, FADE_DURATION);
 }
 
-export function generateBlockList(
-  stxBlocks: (Block | NakamotoBlock)[],
-  btcBlocksMap?: BtcBlockMap
-) {
-  if (stxBlocks.length === 0) return [];
+export function generateBlockList(stxBlocks: (Block | NakamotoBlock)[], btcBlocksMap: BtcBlockMap) {
```
Comment (He1DAr) [inline src/app/_components/BlockList/utils.ts:57]: Would you mind adding a unit test to this function?

File: src/common/queries/useBurnBlock.ts (hunk 3 above)
Comment (BLuEScioN) [reply src/common/queries/useBurnBlock.ts:20]: done

File: src/app/_components/BlockList/utils.ts (hunk 5 above)
Comment (BLuEScioN) [reply src/app/_components/BlockList/utils.ts:57]: sure

File: src/app/_components/BlockList/utils.ts (hunk 4 above)
Comment (BLuEScioN) [reply src/app/_components/BlockList/utils.ts:76]: thanks

**Review by He1DAr — APPROVED** [review]

**Pull request merged**